common/io/src/hts_fasta_parser.cpp

```diff
@@ -34,9 +35,30 @@ namespace claragenomics
 namespace io
 {
 
-FastaParserHTS::FastaParserHTS(const std::string& fasta_file)
+FastaParserHTS::FastaParserHTS(const std::string& fasta_file,
+                               const std::string& output_dir)
 {
-    fasta_index_ = fai_load3(fasta_file.c_str(), NULL, NULL, FAI_CREATE);
+    std::string fai_file = fasta_file + ".fai";
+    std::string gzi_file = fasta_file + ".gzi";
+    if (output_dir.length() != 0)
+    {
+        // resolve idx and gzi file path
+        if (!claragenomics::filesystem::dirExists(output_dir))
+        {
+            throw std::runtime_error("Output dir " + output_dir +
+                                     " not found or not a directory!");
+        }
+
+        std::string file_name =
+            claragenomics::filesystem::resolveFileName(fasta_file);
+        fai_file = output_dir + "/" + file_name + ".fai";
```

Comment:
Is this a good way to be constructing file paths? We should be using `<filesystem>` like [described here](https://stackoverflow.com/a/36848326)